src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -86,6 +117,33 @@ function shouldConstruct(Component) {
   return !!(Component.prototype && Component.prototype.isReactComponent);
 }
 
+// This is used to create an alternate fiber to do work on.
+exports.cloneFiber = function(fiber : Fiber) : Fiber {
+  // We use a double buffering pooling technique because we know that we'll only
+  // ever need at most two versions of a tree. We pool the "other" unused node
```

Comment:
Is there some dev-only (or test-only, even) assertions we could make to ensure that we're not somehow clobbering over other work and this truly acts like a fresh clone? Without that I feel like we are likely to mess something up.